Feature

AlexanderZaborski-tomtom opened the pull request:
Implemented the custom endpoint "api/currentuser".

Frontend not completed yet

AlexanderZaborski-tomtom committed:
Initial upload

AlexanderZaborski-tomtom committed:
Create .DS_Store

AlexanderZaborski-tomtom committed:
Removed unneccessary files

AdamBodera-TomTom commented on custom_endpoint/custom_endpoint.routing.yml:
```diff
@@ -0,0 +1,8 @@
+  defaults:
+    _controller: '\Drupal\custom_endpoint\Controller\CustomEndpointController::renderApi'
+    _title: 'Current user data'
+  methods:  ['GET']
```
unnecessary double space after colon

AlexanderZaborski-tomtom replied:
Removed double space

AdamBodera-TomTom commented on custom_endpoint/custom_endpoint.services.yml:
```diff
@@ -0,0 +1,4 @@
+services:
+  custom_endpoint.currentUser:
+    class: Drupal\custom_endpoint\CustomEndPointController
```
this is not a service, so it is invalid

AlexanderZaborski-tomtom replied:
For some reason old code is showing in this comment.
Now class is pointing to \Drupal\custom_endpoint\CustomEndpointService

AdamBodera-TomTom commented on custom_endpoint/src/Controller/CustomEndpointController.php:
```diff
@@ -0,0 +1,84 @@
+/**
+ * Custom Endpoint Controller class
+ */
+
```
unnecessary empty line

AlexanderZaborski-tomtom replied:
Removed empty space

AdamBodera-TomTom commented on custom_endpoint/src/Controller/CustomEndpointController.php:
```diff
@@ -0,0 +1,84 @@
+      /**
+     * 
+     * @param AccountInterface $currentUserClass constructor
+     */
```
missing method description
parameters should be typed by the full path

constructor comment block should look like this:

```
  /**
   * Constructs a new object.
   *
   * @param \Drupal\Core\Routing\RouteProviderInterface $route_provider
   *   The route provider service.
   */
```

AlexanderZaborski-tomtom replied:
Fixed comment

AdamBodera-TomTom commented:
General comment: 
fix the comment blocks
every method should have it's return type declared - please fix that :)

AdamBodera-TomTom commented on custom_endpoint/src/Controller/CustomEndpointController.php:
```diff
@@ -0,0 +1,84 @@
+         $moddate = $this->currentUser->getLastAccessedTime();
+         $lastlogon = $this->currentUser->getLastAccessedTime();
+
+         if ($username == "") {
```
the check if a user is logged in should be done in routing with requirements field
you can also set a specific permission or a role that the user must have to access the endpoint

AlexanderZaborski-tomtom committed:
Update CustomEndpointController.php

AlexanderZaborski-tomtom committed:
Fixed comment blocks and added some return types

AlexanderZaborski-tomtom committed:
Added all return types

AlexanderZaborski-tomtom committed:
Update CustomEndpointController.php

AlexanderZaborski-tomtom committed:
Removed services.yml

There is no need to inject the current_user services because it is already passed in the create method

AlexanderZaborski-tomtom committed:
Fixed formatting in Controller

AlexanderZaborski-tomtom committed:
Moved get results to service class

AdamBodera-TomTom commented on custom_endpoint/src/Controller/CustomEndpointController.php:
```diff
@@ -0,0 +1,44 @@
+   */
+  public function renderApi(): JsonResponse {
+
+    $currentUserService = \Drupal::service('custom_endpoint.currentUser');
```
now this is a static service invocation - we don't do that
instead dependency inject it the same way as a current_user service

AdamBodera-TomTom commented on custom_endpoint/src/Controller/CustomEndpointController.php:
```diff
@@ -0,0 +1,44 @@
+   */
+  public static function create(ContainerInterface $container): CustomEndpointController {
+    return new static(
+      $container->get('current_user'),
```
move current_user to the CustomEndpointService, it should be injected there

AdamBodera-TomTom commented on custom_endpoint/custom_endpoint.routing.yml:
```diff
@@ -0,0 +1,8 @@
+    _title: 'Current user data'
+  methods:  ['GET']
+  requirements:
+    _access: 'TRUE'
```
this endpoint should not be open - it should be only for the logged in user

AlexanderZaborski-tomtom committed:
Fixed up comments

AlexanderZaborski-tomtom committed:
Removed double space on line 6

AlexanderZaborski-tomtom committed:
Fixed Constructor comment

Added the full path for AccountInterface

AlexanderZaborski-tomtom committed:
Fixed constructor comment

AlexanderZaborski-tomtom committed:
Fixed services.yml

AlexanderZaborski-tomtom committed:
Renamed service

AlexanderZaborski-tomtom committed:
Updated services.yml

Now it points to correct service

AlexanderZaborski-tomtom committed:
Fixed invalid class path in services.yml

AlexanderZaborski-tomtom committed:
Update CustomEndpointController.php
